smartsim/settings/lsfSettings.py

```diff
@@ -535,21 +545,21 @@ def set_queue(self, queue: str) -> None:
         if queue:
             self.batch_args["q"] = queue
 
-    def _format_alloc_flags(self) -> None:
+    def format_alloc_flags(self) -> None:
         """Format ``alloc_flags`` checking if user already
         set it. Currently only adds SMT flag if missing
         and ``self.smts`` is set.
         """
 
         if self.smts:
-            if not "alloc_flags" in self.batch_args.keys():
+            if "alloc_flags" not in self.batch_args.keys():
```

Comment:
This is a functional change, but I'm guessing that the new version is correct and the old version was wrong.

Reply:
I think `not ITEM in COLLECTION` and `ITEM not in COLLECTION` are identical in their result, even if the latter is the preferred style.

Reply:
What's the binding of `not`? Is it `not (item in collection)` or `(not item) in collection`?

Reply:
Reverted change, added unit test `test_bsub_batch_alloc_flag_formatting_by_smt`, repeated change, verified same results.